src/main/java/game/map/Map.java

```diff
@@ -77,4 +80,42 @@ public List<Player> getPlayers() {
         return players;
     }
 
-}
+        public static HashMap<Player, List<Country>> assignCountries(List<Player> players, List<Country> countries) {
+            HashMap<Player, List<Country>> playerCountryMap = new HashMap<>();
+
+            // Check if the countries list is empty
+            if (countries.isEmpty()) {
+                System.out.println("No countries available to assign.");
+                return playerCountryMap;
+            }
+
+            // Shuffle the countries
+            Collections.shuffle(countries);
+
+            // Calculate the number of countries each player should have
+            int numCountriesPerPlayer = 5;
+
+            // Assign countries to players
+            int currentIndex = 0;
+            for (Player player : players) {
+                List<Country> assignedCountries = new ArrayList<>();
+                for (int i = 0; i < numCountriesPerPlayer && currentIndex < countries.size(); i++) {
+                    assignedCountries.add(countries.get(currentIndex));
+                    currentIndex++;
+                }
+                player.setAssignedCountries(assignedCountries);
+                playerCountryMap.put(player, assignedCountries);
+            }
+
+            // Assign remaining countries to players randomly
+            Random random = new Random();
+            while (currentIndex < countries.size()) {
```

Comment:
The idea is good, but here if the Random generator generates the same random number all the times, then you would assign all the remaining countries to the same player.

I think each remaining country should be assigned to a different player. So a player would get only 1 extra country than any other player. You can maybe use the below code

         ```while (currentIndex < countries.size()) {
                List<Integer> playerIndices = IntStream.range(0, players.size()).boxed().collect(toList());
                Collections.shuffle(playerIndices);
                int idxInPlayerIndices = 0;
                while (currentIndex < countries.size()) {
                    Player randomPlayer = players.get(playerIndices.get(idxInPlayerIndices));
                    randomPlayer.getCountries().add(countries.get(currentIndex));
                    idxInPlayerIndices ++;
                    currentIndex++;
                }
            }```